### ISSUE TITLE
Parse special property @description.

### ISSUE THREAD

**Issue by 7tomek** [icanzilb/JSONModel#418]
Hi,

I found bug while parsing json data. When we have special name of property in json (@description). Please look on my repo https://github.com/7tomek/JSONModelTests there are patch with test and lldb-console (print object).

**Comment by billinghamj** [comment]
Hi @7tomek. Rather than creating a patch file, could you fork this repo and commit the changes directly to that? Makes it much easier for us to work with your changes.

**Comment by 7tomek** [comment]
https://github.com/7tomek/JSONModel/commit/026d32de0070c795c8cd14ad7682e8372a94fa78

**Comment by billinghamj** [comment]
Thanks for that. Okay, so basically you've stumbled upon a feature of KVC in Obj-C.

See: https://stackoverflow.com/a/20453438

`@` at the start of a dictionary key when calling `keyPath` methods (as JSONModel does) has special meaning. This functionality allows JSONModel to support patterns like this:

https://github.com/icanzilb/JSONModel/blob/master/JSONModelDemoTests/UnitTests/TestModels/ExtremeNestingModel.m#L17

As such, we can't remove or change this functionality. We might be able to escape or handle specifically the `@` operators though.

**Comment by billinghamj** [comment]
Looks like the operators cannot be escaped. As such, we will not be able to resolve this without removing the `keyPath` feature.

Unless you have any other suggestions, I suspect we simply won't be able to fix this one I'm afraid...

**Comment by Johnykutty** [comment]
This is not for all special characters(I believe) but surely for @. More precisely keys starting with @.
My suggestion is to do something like send the dictionary and key to the Model class itself and it can pick the value it and with `objectForKey:` method instead of `valueForKeyPath:`

@billinghamj  your thoughts?

**Comment by billinghamj** [comment]
Yes, it is just keys starting with `@` which are be the problem.

One possible option would be allowing models to opt-out of the `valueForKeyPath:` methods in favor of `objectForKey:`, losing the key path functionality.

**Comment by billinghamj** [comment]
I'm inclined to say that this should be resolved in a JSONModel v2 which allows for more explicit configuration via the implementation file.
